### PR TITLE
Tanks

### PR DESCRIPTION
**Pull request by IndDevComp**

### CODE REVIEW

File: Tanks/Tanks/Engine/Commands/Rotate.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿using Tanks.Engine.EntityProp;
+
+namespace Tanks.Engine.Commands
+{
+    public class Rotate<T> : Command
```
Comment (IndDevComp) [inline Tanks/Tanks/Engine/Commands/Rotate.cs:5]: T лишний, забыл удалить

File: Tanks/Tanks/Engine/Command.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Tanks.Engine.Commands
+{
+    public interface Command
```
Comment (IndDevComp) [inline Tanks/Tanks/Engine/Command.cs:3]: sdasdadasdsa

File: Tanks/TanksUnitTest/TestingGameEngine.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            Assert.AreEqual(2, tank.Object.Velocity);
+        }
+        [TestMethod]
+        public void test_TankShouldRotateWithSomeVelocity()
+        {
+            var tank = new Mock<Rotatable>();
+            tank.Setup(a => a.Angle).Returns(2);
+            tank.Setup(a => a.Velocity).Returns(5);
+
+            Rotate cmd = new Rotate();
```
Comment (etyumentcev) [inline Tanks/TanksUnitTest/TestingGameEngine.cs:34]: new Rotate(tank.Object)

File: Tanks/TanksUnitTest/TestingGameEngine.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            double result = cmd.Action(tank.Object);
```
Comment (etyumentcev) [inline Tanks/TanksUnitTest/TestingGameEngine.cs:35]: cmd.Action()

Comment (etyumentcev) [reply Tanks/TanksUnitTest/TestingGameEngine.cs:35]: void Action()

Comment (etyumentcev) [reply Tanks/TanksUnitTest/TestingGameEngine.cs:35]: tank.SetupSet(a => a.Angle = 7).Verifiable();

File: Tanks/TanksUnitTest/TestingGameEngine.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+            Assert.AreEqual(7, result);
```
Comment (etyumentcev) [inline Tanks/TanksUnitTest/TestingGameEngine.cs:37]: удалить, так как tank.SetupSet(a => a.Angle = 7).**Verifiable();**

File: Tanks/TanksUnitTest/TestingGameEngine.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            Mock<Rotatable> tank = new Mock<Rotatable>();
+            tank.Setup(a => a.Velocity).Returns(2);
+
+            Assert.AreEqual(2, tank.Object.Velocity);
+        }
+        [TestMethod]
+        public void test_TankShouldRotateWithSomeVelocity()
+        {
+            var tank = new Mock<Rotatable>();
+            tank.Setup(a => a.Angle).Returns(2);
```
Comment (etyumentcev) [inline Tanks/TanksUnitTest/TestingGameEngine.cs:31]: tank.Setup(a => a.Angle).Returns(2)**.Verifiable()**;

File: Tanks/TanksUnitTest/TestingGameEngine.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+            tank.Setup(a => a.Velocity).Returns(5);
+
+            Rotate cmd = new Rotate();
+            double result = cmd.Action(tank.Object);
+
+            Assert.AreEqual(7, result);
+        }
```
Comment (etyumentcev) [inline Tanks/TanksUnitTest/TestingGameEngine.cs:38]: var tank = new Mock<Rotatable>();
            tank.Setup(a => a.Angle).Returns(2).Verifiable();
            tank.Setup(a => a.Velocity).Returns(5).Verifiable();
            tank.SetupSet(a => a.Angle = 7).Verifiable();
            
            Rotate cmd = new Rotate(tank.Object);
            cmd.Action();

Comment (etyumentcev) [reply Tanks/TanksUnitTest/TestingGameEngine.cs:38]: tank.Verify();

**Pull request closed without merging**